### PR TITLE
add unified_mode true

### LINKED ISSUE #484: Add unified_mode for Chef 17 support
Please add unified_mode to the resources so that this cookbook supports chef 17

### PR DESCRIPTION
**Pull request by derekgroh**
# Description

Adds cookbook compatibility with Chef 17

## Issues Resolved

#484 

## Check List

- [X] All tests pass. See TESTING.md for details.
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: kitchen.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ transport:
 
 provisioner:
   name: chef_zero
-  deprecations_as_errors: true
+  deprecations_as_errors: false
```
Comment (ramereth) [inline kitchen.yml:13]: I'm pretty sure we can remove the dependency on the windows cookbook if we're bumping unified mode which would fix this. Can you please do the following?
- Revert this to `true`
- Remove the windows dependency in `metadata.rb`
- Bump the required chef version to `>= 15.3` in `metadata.rb`

**Review by ramereth — DISMISSED** [review]

File: kitchen.yml (hunk 1 above)
Comment (derekgroh) [reply kitchen.yml]: Pretty close, manager looks like the only resource that needs love.

File: resources/app.rb [hunk 2]
```diff
@@ -17,6 +17,8 @@
 # limitations under the License.
 #
 
+unified_mode true if respond_to?(:unified_mode)
```
Comment (damacus) [inline resources/app.rb:20]: Please revert this change. Having it optionally on invalidates testing.

**Review by damacus — CHANGES_REQUESTED** [review]

**Review by damacus — APPROVED** [review]
❤️ 🥳 
LGTM?

**Comment by kitchen-porter** [comment]
Released as: [8.0.0](https://github.com/sous-chefs/iis/releases/tag/8.0.0)
